mkosi/distributions/arch.py

```diff
@@ -69,9 +70,24 @@ def repositories(cls, context: Context) -> Iterable[PacmanRepository]:
             yield PacmanRepository("core", context.config.local_mirror)
         else:
             if context.config.architecture.is_arm_variant():
-                url = f"{context.config.mirror or 'http://mirror.archlinuxarm.org'}/$arch/$repo"
+                if context.config.snapshot and not context.config.mirror:
+                    die("There is no known public mirror for snapshots of Arch Linux ARM")
+
+                mirror = context.config.mirror or "http://mirror.archlinuxarm.org"
+            else:
+                if context.config.mirror:
+                    mirror = context.config.mirror
+                elif context.config.snapshot:
+                    mirror = "https://archive.archlinux.org"
```

Comment:
We also do have mirrors for the archive, therefore it would be good to not only hit on the main mirror:
```ini
[archive_mirrors]
america.mirror.pkgbuild.com
asia.mirror.pkgbuild.com
europe.mirror.pkgbuild.com
```

We might need to introduce something like `geo.mirror.pkgbuild.com` for the archive though 🤔

Reply:
@christian-heusel I don't think we want to hardcode mirrors like this in mkosi, a geo mirror for the archive would be much appreciated. For now I dropped the code that makes use of the archive in CI, but I'm going to merge this with the official mirror for now. Happy to switch to a geo mirror url when that becomes available.